src/java/org/apache/cassandra/config/Config.java

```diff
@@ -261,82 +297,102 @@
 
     // Commit Log
     public String commitlog_directory;
-    public Integer commitlog_total_space_in_mb;
+    @Replaces(oldName = "commitlog_total_space_in_mb", converter = Converter.MegabytesDataStorageConverter.class, deprecated = true)
+    public DataStorage commitlog_total_space;
     public CommitLogSync commitlog_sync;
 
     /**
      * @deprecated since 4.0 This value was near useless, and we're not using it anymore
      */
     public double commitlog_sync_batch_window_in_ms = Double.NaN;
-    public double commitlog_sync_group_window_in_ms = Double.NaN;
-    public int commitlog_sync_period_in_ms;
-    public int commitlog_segment_size_in_mb = 32;
+    @Replaces(oldName = "commitlog_sync_group_window_in_ms", converter = Converter.MillisDurationInDoubleConverter.class, deprecated = true)
```

Comment:
Should we handle `Nan` in `MillisDurationInDoubleConverter` and convert it to `0ms`?

Reply:
You mean in case someone hooks directly to `Config` and use it or? I don't see a big benefit but I might be wrong.

Reply:
It was the old default, but I guess you're saying that an actual config file couldn't have contained it, so it doesn't really matter?

Reply:
Yes.
As this was a change done long time ago I also did a quick test in trunk in case I forgot something. Here is the result:
```Exception (org.apache.cassandra.exceptions.ConfigurationException) encountered during startup: Invalid yaml: 
file:/Users/ekaterina.dimitri/IdeaProjects/april/cassandra/conf/cassandra.yaml
 Error: Cannot create property=commitlog_sync_group_window_in_ms for JavaBean=org.apache.cassandra.config.Config@34e9fd99
 in 'reader', line 11, column 1:
    cluster_name: 'Test Cluster'
    ^
For input string: "Double.NaN"
 in 'reader', line 456, column 36:
    commitlog_sync_group_window_in_ms: Double.NaN
```

Reply:
Annnnd It seems I am wrong, `Nan` works....

Reply:
Good catch, I will add that, thank you!

Reply:
Looking into CASSANDRA-12199 I am wondering whether it really makes sense to be introduced.... 
I think it was added no express no value, but not really with the expectation someone will set it to `Nan`